pacman/model/partitioner_splitters/abstract_splitters/abstract_splitter_common.py

```diff
@@ -0,0 +1,282 @@
+        else:
+            # Currently on a max so
+            if is_fixed_atoms:
+                # As new is fixed max sure it is not higher than max
+                if max_atoms_per_core > self._max_atoms_per_core:
+                    raise PacmanConfigurationException(
+                        self.FIXED_ABOVE_MAX.format(
+                            max_atoms_per_core, self._max_atoms_per_core))
+                else:  # Set the new fixed
+                    self._max_atoms_per_core = max_atoms_per_core
+                    self._is_fixed_atoms_per_core = True
+            else:
+                # Both max so only change if new max if lower
+                if max_atoms_per_core < self._max_atoms_per_core:
+                    # Set the new max but leave fixed false
+                    self._max_atoms_per_core = max_atoms_per_core
+                else:
+                    return  # Ok to Ignore a higher or same max
+
+    @property
+    def governed_app_vertex(self):
+        return self._governed_app_vertex
+
+    def set_governed_app_vertex(self, app_vertex):
+        """ sets a app vertex to be governed by this splitter object. \
+            Once set it can't be reset
+
+        :param ApplicationVertex app_vertex: the app vertex to govern
+        :raises PacmanConfigurationException: \
+            if the app vertex has already been set.
+        """
+        if self._governed_app_vertex == app_vertex:
+            return
+        if self._governed_app_vertex is not None:
+            raise PacmanConfigurationException(
+                self.SETTING_SPLITTER_ERROR_MSG.format(
+                    self._governed_app_vertex, self._splitter_name,
+                    app_vertex))
+        self._governed_app_vertex = app_vertex
+        self.check_supported_constraints()
+        app_vertex.splitter = self
+
+    def check_supported_constraints(self):
+        utility_calls.check_algorithm_can_support_constraints(
+            constrained_vertices=[self._governed_app_vertex],
+            supported_constraints=[
+                MaxVertexAtomsConstraint, FixedVertexAtomsConstraint],
+            abstract_constraint_type=AbstractPartitionerConstraint)
+
+    def split(self, resource_tracker, machine_graph):
+        """ executes splitting
+
+        :param ResourceTracker resource_tracker: machine resources
+        :param MachineGraph machine_graph: machine graph
+        :return: true if successful, false otherwise
+        :rtype: bool
+        """
+        return self.create_machine_vertices(resource_tracker, machine_graph)
+
+    @abstractmethod
+    def create_machine_vertices(self, resource_tracker, machine_graph):
+        """ method for specific splitter objects to use.
+
+        :param resource_tracker:
+        :param machine_graph:
+        :return: true if successful, false otherwise
+        :rtype: bool
+        """
+
+    @abstractmethod
+    def get_out_going_slices(self):
```

Comment:
pre and post were confusing as heck, to the point i was getting confused. As pre was the pre of the edge, but most of the time was the outgoing of the vertex we're discussing. and post was actually the verts that receive stuff. so was giving me headaches. this we think is better. as its using vertex scoped terms, instead of edge scoped terms